scanner/src/main/kotlin/ScanContext.kt

```diff
@@ -34,5 +35,10 @@ data class ScanContext(
     /**
      * The [type][PackageType] of the packages to scan.
      */
-    val packageType: PackageType
+    val packageType: PackageType,
+
+    /**
+     * The [RepositoryConfiguration] of the project to scan.
+     */
+    val repositoryConfiguration: RepositoryConfiguration
```

Comment:
The `RepositoryConfiguration` contains many unrelated properties, like the analyzer configuration, resolutions, or curations. Therefore it would be better to only pass the `Excludes`.